Pod/Classes/BONTag.m

```diff
@@ -0,0 +1,235 @@
+
+    // Iterate over the string finding each escape in order until there are no more escape strings
+    while (YES) {
+        BONTag *nextTag;
+        NSRange nextEscapeRange;
+        NSRange nextEscapedTagRange;
+
+        // Find the next escape string
+        for (BONTag *tag in tags) {
+            NSRange escapeRange = [theString rangeOfString:tag.escapeString options:0 range:searchRange];
+
+            if (escapeRange.location != NSNotFound) {
+                if (!nextTag || (escapeRange.location < nextEscapeRange.location)) {
+                    // Check if this character is escaping a start or end tag
+                    for (NSString *tagString in @[ tag.startTag, tag.endTag ]) {
+                        // Check if there is room for this tag to exist after the escape character
+                        if ((NSInteger)(NSMaxRange(escapeRange) + tagString.length) < theString.length) {
+                            // Check if the following characters are actually the tag
+                            NSRange potentialEscapedTagRange = NSMakeRange(NSMaxRange(escapeRange), tagString.length);
+                            if ([[theString substringWithRange:potentialEscapedTagRange] isEqualToString:tagString]) {
+                                nextTag = tag;
+                                nextEscapeRange = escapeRange;
+                                nextEscapedTagRange = potentialEscapedTagRange;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        if (!nextTag) {
+            break;
+        }
+
+        NSRange range = NSMakeRange(nextEscapeRange.location, nextEscapedTagRange.length + nextEscapeRange.length);
+
+        if (stripEscapeCharacters) {
+            theString = [theString stringByReplacingOccurrencesOfString:nextTag.escapeString withString:@"" options:0 range:nextEscapeRange];
+            range = NSMakeRange(nextEscapeRange.location, nextEscapedTagRange.length);
+        }
+
+        [escapedRanges addObject:[NSValue valueWithRange:range]];
+
+        searchRange = NSMakeRange(NSMaxRange(range), [theString length] - NSMaxRange(range));
+    }
+
+    *string = theString;
+
+    return escapedRanges;
+}
+
++ (NSRange)findNextString:(NSString *)string inString:(NSString *)stringToSearch ignoringRanges:(NSArray *)escapedRanges range:(NSRange)range
+{
+    NSRange searchRange = range;
+
+    while (YES) {
+        NSRange stringRange = [stringToSearch rangeOfString:string options:0 range:searchRange];
+
+        // Ignore this match
+        if ([escapedRanges containsObject:[NSValue valueWithRange:stringRange]]) {
+            searchRange = NSMakeRange(NSMaxRange(stringRange), [stringToSearch length] - NSMaxRange(stringRange));
+            continue;
+        }
+
+        return stringRange;
+    }
+}
+
++ (NSArray *)rangesInString:(NSString **)string betweenTags:(NSArray *)tags stripTags:(BOOL)stripTags
```

Comment:
Good question, there really is no reason I can think of. Will remove this code path.